modules/bmc/redfish/dell.rb

```diff
@@ -0,0 +1,18 @@
+module RedfishVendorOverridesDellInc
+
+  def powercycle
+    # ForceRestart was added in Lifecycle Controller 3.30.30.30
+    # prior to that, it required ForceOff followed by On.
+    if host.System.Members[0].Actions['#ComputerSystem.Reset']['ResetType@RedfishAllowableValues'].include? 'ForceRestart'
+      poweraction('ForceRestart')
+    else
+      poweraction('ForceOff')
+      # REVIEW - I wonder if any kind of delay is needed for this to be reliable.
```

Comment:
Anyway to test this? probably in the grand scheme of things (e.g. how long it actually take to reboot), adding some delay here won't hurt.

Reply:
I wanted to resolve this question, but the way my access to real BMCs and dev/test instance of Foreman shook out meant I couldn't do it. Theoretically the iDRAC will queue the events, but _theoretically_ the iDRAC will do all _kinds_ of things it doesn't actually do. A sleep may not be sufficient, in case it doesn't -- it may require using the API to check status of "long running" Redfish actions. Perhaps it's better just to not support the older iDRAC without ForceRestart, in this initial commit. I'm open to direction.

Reply:
are you worried about a race condition? or simply ignoring the older command (so its on and stays on?)
I think its acceptable to power off, busy wait and turn on again, I'm sure folks with older hardware would still prefer an option vs no option at all? (assuming its not too much of an effort..?)

Reply:
Power state changes aren't exposed as asynchronous redfish calls (at least not on Dell), so we can't use that. It also turns out the Dells are more restrictive about how they accept power state change actions and they weren't working at all the way I had it coded, so it's a good thing I went back to explicitly test.